osu.Game/Overlays/Profile/Sections/RanksSection.cs

```diff
@@ -1,12 +1,199 @@
+                        best.Add(new DrawableScore(score, Math.Pow(0.95, i))
+                        {
+                            RelativeSizeAxes = Axes.X,
+                            Height = 60,
+                        });
+                        i++;
+                    }
+                }
+                best.ShowMore();
+            }
+        }
+
+        public Score[] ScoresFirst
+        {
+            set
+            {
+                first.Clear();
+                if (value.Length == 0)
+                {
+                    firstMissing.Show();
+                }
+                else
+                {
+                    firstMissing.Hide();
+                    foreach (Score score in value)
+                        first.Add(new DrawableScore(score)
+                        {
+                            RelativeSizeAxes = Axes.X,
+                            Height = 60,
+                        });
+                }
+                first.ShowMore();
+            }
+        }
+
+        private class ScoreFlowContainer : Container<DrawableScore>
+        {
+            private readonly FillFlowContainer<DrawableScore> scores;
+            private readonly OsuClickableContainer showMoreText;
+
+            protected override Container<DrawableScore> Content => scores;
+
+            private int shownScores;
+
+            public ScoreFlowContainer()
+            {
+                InternalChild = new FillFlowContainer
+                {
+                    AutoSizeAxes = Axes.Y,
+                    RelativeSizeAxes = Axes.X,
+                    Direction = FillDirection.Vertical,
+                    Children = new Drawable[]
+                    {
+                        scores = new FillFlowContainer<DrawableScore>
+                        {
+                            AutoSizeAxes = Axes.Y,
+                            RelativeSizeAxes = Axes.X,
+                            Direction = FillDirection.Vertical,
+                        },
+                        showMoreText = new ShowMoreContainer
+                        {
+                            Action = ShowMore,
+                            AutoSizeAxes = Axes.Both,
+                            Anchor = Anchor.TopCentre,
+                            Origin = Anchor.TopCentre,
+                            Alpha = 0,
+                            Child = new OsuSpriteText
+                            {
+                                TextSize = 14,
+                                Text = "show more",
+                            }
+                        }
+                    },
+                };
+            }
+
+            public override void Clear(bool disposeChildren)
+            {
+                base.Clear(disposeChildren);
+                shownScores = 0;
+                showMoreText.Show();
+            }
+
+            public void ShowMore()
+            {
+                shownScores = Math.Min(Children.Count, shownScores + 5);

```

Comment:
This method could be written a lot nicer (you shouldn't need the `shownScores` field at all).